cdap-examples/DataCleansing/src/main/java/co/cask/cdap/examples/datacleansing/DataCleansingMapReduce.java

```diff
@@ -0,0 +1,156 @@
+import org.apache.hadoop.mapreduce.Reducer;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * A simple MapReduce that reads records from the inputRecords PartitionedFileSet and writes all records
+ * that match a particular {@link Schema} to the outputRecords PartitionedFileSet. It also keeps track of its state of
+ * which partitions it has processed, so that it only processes new partitions of data each time it runs.
+ */
+public class DataCleansingMapReduce extends AbstractMapReduce {
+  protected static final String NAME = "DataCleansingMapReduce";
+  protected static final String OUTPUT_PARTITION_KEY = "output.partition.key";
+  protected static final String SCHEMA_KEY = "schema.key";
+
+  private final BatchPartitionConsumer partitionConsumer = new BatchPartitionConsumer() {
+    private static final String STATE_KEY = "state.key";
+
+    @Nullable
+    @Override
+    protected byte[] readBytes(DatasetContext datasetContext) {
+      return ((KeyValueTable) datasetContext.getDataset(DataCleansing.CONSUMING_STATE)).read(STATE_KEY);
+    }
+
+    @Override
+    protected void writeBytes(DatasetContext datasetContext, byte[] stateBytes) {
+      ((KeyValueTable) datasetContext.getDataset(DataCleansing.CONSUMING_STATE)).write(STATE_KEY, stateBytes);
+    }
+  };
+
+  @Override
+  public void configure() {
+    setName(NAME);
+    setMapperResources(new Resources(1024));
+    setReducerResources(new Resources(1024));
+  }
+
+  @Override
+  public void beforeSubmit(MapReduceContext context) throws Exception {
+    PartitionedFileSet inputRecords = partitionConsumer.getConfiguredDataset(context, DataCleansing.INPUT_RECORDS);
+    context.setInput(DataCleansing.INPUT_RECORDS, inputRecords);
+
+    // Each run writes its output to a partition for the league
+    Long timeKey = Long.valueOf(context.getRuntimeArguments().get(OUTPUT_PARTITION_KEY));
+    PartitionKey outputKey = PartitionKey.builder().addLongField("time", timeKey).build();
+    Map<String, String> outputArgs = new HashMap<>();
+    PartitionedFileSetArguments.setOutputPartitionKey(outputArgs, outputKey);
+
+    PartitionedFileSet outputFileSet = context.getDataset(DataCleansing.OUTPUT_RECORDS, outputArgs);
+    context.setOutput(DataCleansing.OUTPUT_RECORDS, outputFileSet);
+
+    Job job = context.getHadoopJob();
+    job.setMapperClass(SchemaMatchingFilter.class);
+    job.setReducerClass(IdentityReducer.class);
+    job.setNumReduceTasks(1);
+
+    // simply propagate the schema (if any) to be used by the mapper
+    String schemaJson = context.getRuntimeArguments().get(SCHEMA_KEY);
+    if (schemaJson != null) {
+      job.getConfiguration().set(SCHEMA_KEY, schemaJson);
+    }
+  }
+
+  @Override
+  public void onFinish(boolean succeeded, MapReduceContext context) throws Exception {
+    if (succeeded) {
+      partitionConsumer.persist(context);
+    }
+    super.onFinish(succeeded, context);
```

Comment:
But does it do anything in the super class? Otherwise its just a no-op method call?